students/CarolF/session04/MailRoom_2.py

```diff
@@ -0,0 +1,125 @@
+
+    if response ==4:
+        print("Goodbye")
+        exit() 
+
+    if response ==3:
+        print("Send Letters to All")
+        sendLettersToAll()      
+
+    if response ==2:
+        createReportData()	
+
+    if response == 1: 
+        findDonorName()
+
+def findDonorName():
+    """This function identifies the donor information to be used"""
+    donorName = str(input("Please provide the donor name for the thank you or type 'list' to see the list of donors >"))
+
+    while donorName == 'list': #as long as user types list, it will return the donor list
+        for key in donorList:  print(key) #dictionary version to provide list
+        donorName = str(input("Please enter the donor name to compose the thank you >"))
+    
+    if(donorName not in donorList):
+        isNewDonor = True 
+        print(f'Adding {donorName} as new donor to list') #remove after debugging
+        retrieveDonationAmount(donorName, isNewDonor)
+        
+    elif (donorName in donorList):
+        isNewDonor = False
+        print("found! ", donorName)
+        retrieveDonationAmount(donorName, isNewDonor)
+
+
+
+def retrieveDonationAmount(donorName, isNewDonor):
+    """requests donor information and will append new contibution to former donor"""
+    newContribution = float(input("please enter new contribution amount....> "))
+    if isNewDonor is False:
+         newListValue = donorList[donorName] 
+         donorList[donorName].append(newContribution)       
+         printLetter(donorName,newContribution)
+    else:
+        donorList[donorName] = [newContribution] #turned value into single value list in anticipation of adding contributions in the future
+        printLetter(donorName,newContribution)
+
+    print(donorList)
+
+def printLetter(donorName, newContribution):
+    """completes and prints Thank you letter then calls getAction method to request new action"""
+    print("\nDear {},".format(donorName)) #added newlines to improve readability
+    print("Thank you for your generous contribution of ${} ".format(newContribution))
+    print("Your contribution is greatly appreicated and will be used to grow our operations and will reinforce our commitment to innovation.")
+    print("Sincerely,")
+    print("The Entire Team at PyCert.com\n")
+    getAction(usersChoice)
+
+
+#still need to call and improve formattting.
+def createReportData():
+    print('{:<15}'.format("Donor Name"), '{:<20}'.format('|  Total Given  |'),
+                 '{:>5}'.format('Number of Gifts  |'),'{:>20}'.format('Average Gift  |')) 
+    print('_'*77)
+    dataForDonationReport = []
+    for key in donorList:
+        donorNumTotal = sum(donorList[key])
+        numbGifts = (len(donorList[key]))
+        donorAverage = "$ {0:.2f}".format(float((sum(donorList[key]))/numbGifts))
+        dataForDonationReport.append([donorNumTotal,key,numbGifts,donorAverage]) 
+    
+    sortedReport = sorted(dataForDonationReport)
+    ascendingReport =sortedReport[::-1]
+    
```

Comment:
I would remove all of the print stuff here, and put them in their own function. Makes this function cleaner and smaller and easier to test.